uber/models.py

```diff
@@ -759,35 +766,43 @@ def auto_badge_num(self, badge_type):
             a specific range.
             :return:
             """
-            in_range = self.query(Attendee.badge_num).filter(Attendee.badge_num >= c.BADGE_RANGES[badge_type][0],
-                                                             Attendee.badge_num <= c.BADGE_RANGES[badge_type][1])
-            if in_range.count():
-                in_range_list = [int(row[0]) for row in in_range.order_by(Attendee.badge_num).all()]
+            in_range = self.query(Attendee.badge_num).filter(
+                Attendee.badge_num != None,
+                Attendee.badge_num >= c.BADGE_RANGES[badge_type][0],
+                Attendee.badge_num <= c.BADGE_RANGES[badge_type][1])
 
+            in_range_list = [int(row[0]) for row in in_range.order_by(Attendee.badge_num).all()]
+            if len(in_range_list):
                 # Searches badge range for a gap in badge numbers; if none found, returns the latest badge number + 1
                 # Doing this lets admins manually set high badge numbers without filling up the badge type's range.
                 start, end = c.BADGE_RANGES[badge_type][0], in_range_list[-1]
                 gap_nums = sorted(set(range(start, end + 1)).difference(in_range_list))
                 if not gap_nums:
-                    return in_range.order_by(Attendee.badge_num.desc()).first().badge_num + 1
+                    return end + 1
                 else:
                     return gap_nums[0]
             else:
                 return c.BADGE_RANGES[badge_type][0]
 
-        def shift_badges(self, badge_type, badge_num, *, until=None, **direction):
-            # assert_badge_locked()
-            until = until or c.BADGE_RANGES[badge_type][1]
+        def shift_badges(self, badge_type, badge_num, *, until=None, up=False, down=False):
             if not c.SHIFT_CUSTOM_BADGES or c.AFTER_PRINTED_BADGE_DEADLINE:
                 return False
-            assert not any(param for param in direction if param not in ['up', 'down']), 'unknown parameters'
-            assert len(direction) < 2, 'you cannot specify both up and down parameters'
-            down = (not direction['up']) if 'up' in direction else direction.get('down', True)
-            shift = -1 if down else 1
-            for a in self.query(Attendee).filter(Attendee.badge_num is not None,
```

Comment:
Calling a filter with `Attendee.badge_num is not None` will not actually do anything. In order to test for `NULL` values you must use `==` or `!=` instead of `is` or `is not`.

`Attendee.badge_num is not None` will evaluate to `True`, so it's equivalent to doing
```python
self.query(Attendee).filter(True,
        Attendee.badge_num >= badge_num,
        Attendee.badge_num <= until)
```